mm2src/coins/utxo/utxo_common.rs

```diff
@@ -2019,19 +2019,99 @@ pub async fn send_maker_refunds_payment<T: UtxoCommonOps + SwapOps>(
+    index: usize,
+    signature_version: SignatureVersion,
+    fork_id: u32,
+) -> Result<bool, String> {
+    // Extract the signature from the scriptSig.
+    let signature = extract_signature(script)?;
+    // Validate the signature.
+    try_s!(SecpSignature::from_der(&signature[..signature.len() - 1]));
+    let signature = signature.into();
+    // Make sure we have no more instructions. P2PK scriptSigs consist of a single instruction only containing the signature.
+    if script.get_instruction(1).is_some() {
+        return ERR!("Unexpected instruction at position 2 of script {:?}", script);
     };
+    // Get the scriptPub for this input. We need it to get the transaction sig_hash to sign (but actually "to verify" in this case).
+    let pubkey = expected_pubkey
+        .to_secp256k1_pubkey()
+        .map_err(|e| ERRL!("Error converting plain pubkey to secp256k1 pubkey: {}", e))?;
+    // P2PK scriptPub has two valid possible formats depending on whether the public key is written in compressed or uncompressed form.
+    let possible_pubkey_scripts = [
+        Builder::build_p2pk(&Public::Compressed(pubkey.serialize().into())),
+        Builder::build_p2pk(&Public::Normal(pubkey.serialize_uncompressed().into())),
+    ];
+    for pubkey_script in possible_pubkey_scripts {
+        // Get the transaction hash that has been signed in the scriptSig.
+        let hash = match signature_hash_to_sign(
+            unsigned_tx,
+            index,
+            &pubkey_script,
+            // FIXME: But P2PK scripts never use segwit as signature version. Should we hardcode this to SignatureVersion::Base or ::ForkId?
+            //        UPD: I think we can safely remove this fixme if the following reasoning holds:
+            //        Since we are using p2pk here, this means the coin isn't segwit. right? but it's the coin supplied from the caller anyways
+            //        inside `check_all_utxo_inputs_signed_by_pub` and we can use the signature version it uses (which isn't segwit for sure).
+            //        But: why is it true that a segwit coin will never produce p2pk spends? (check with_key_pair.rs - sign_tx(), it doesn't reject signing a p2pk input)
+            signature_version,
+            SIGHASH_ALL,
+            fork_id,
+        ) {
+            Ok(hash) => hash,
+            Err(e) => return ERR!("Error calculating signature hash: {}", e),
+        };
+        // Verify that the signature is valid for the transaction hash with respect to the expected public key.
+        return match expected_pubkey.verify(&hash, &signature) {
+            Ok(is_successful) => {
+                if is_successful {
+                    Ok(true)
+                } else {
+                    // If the verification isn't successful, try the other possible pubkey script.
+                    continue;
+                }
+            },
+            Err(e) => ERR!("Error verifying signature: {}", e),
+        };
```

Comment:
nit: would you like to have this code as:
```suggestion
        return match expected_pubkey.verify(&hash, &signature) {
            Ok(true) => Ok(true),
            Ok(false) => continue, // The signature is invalid for this public key. Try the other possible public key script.
            Err(e) => ERR!("Error verifying signature: {}", e),
        };
```

Reply:
thanks [3f6ce9b](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2408/commits/3f6ce9b534eff5a05cb20556767e52cf99bbc779)